orbotservice/src/main/java/org/torproject/android/service/OrbotService.java

```diff
@@ -1628,10 +1622,8 @@ else if (proxyPass != null)
 
                 if (bridgeList != null && bridgeList.length() > 5) //longer then 1 = some real values here
                 {
-                    String[] bridgeListLines = bridgeList.trim().split("\\n");
-
-
-                    int bridgeIdx = (int)Math.round(Math.random()*((double)bridgeListLines.length));
+                    String[] bridgeListLines = parseBridgesFromSettings(bridgeList);
+                    int bridgeIdx = (int)Math.floor(Math.random()*((double)bridgeListLines.length));
```

Comment:
This random used to have the chance to cause an index out of bounds exception prior to this PR. I'm not sure though why we select one bridge instead of loading them all into torrc

Reply:
This was meant for use with built-in bridges, to reduce the chance that a censor-state would block all the bridges at once if it saw traffic to them. It also reduces load on the built-in bridges, so that Tor users don't all connect to them in one fixed order. We need a smarter way to implement this than was done before, and this should really only be done with built-in bridges and not user entered ones, anyhow.